net.solarnetwork.node.weather.ibm.wc/src/net/solarnetwork/node/weather/ibm/wc/BasicWCClient.java

```diff
@@ -0,0 +1,282 @@
+	 * week. The more detailed features in "daypart" such as cloud cover are
+	 * split into day and night, meaning two entries for each day.
+	 */
+	private BigDecimal parseBigDecimal(JsonNode n) {
+		if ( n.isNull() ) {
+			return null;
+		}
+		return new BigDecimal(n.asLong());
+	}
+
+	private LocalTime parseTimeValue(JsonNode n) {
+		if ( n.isNull() ) {
+			return null;
+		}
+		return new LocalTime(n.asLong() * 1000);
+
+	}
+
+	private Integer parseInt(JsonNode n) {
+		if ( n.isNull() ) {
+			return null;
+		}
+		return n.asInt();
+	}
+
+	//TODO valid time UTC
+	@Override
+	public Collection<GeneralDayDatum> readDailyForecast(String locationIdentifier, String apiKey,
+			String datumPeriod) {
+
+		if ( locationIdentifier == null ) {
+			return null;
+		}
+		final List<GeneralDayDatum> result = new ArrayList<GeneralDayDatum>();
+		final String url = getURL(this.dailyForecastUrl, locationIdentifier, apiKey, datumPeriod);
+		JsonNode root;
+		try {
+			log.debug("opening IBM Weather API connection");
+			log.debug(url);
+			URLConnection conn = getURLConnection(url, HTTP_METHOD_GET);
+			root = getObjectMapper().readTree(getInputStreamFromURLConnection(conn));
+		} catch ( IOException e ) {
+			log.warn("Error reading WC URL [{}]: {}", url, e.getMessage());
+			return result;
+		}
+
+		for ( JsonNode n : root.get("dayOfWeek") ) {
+			result.add(new GeneralDayDatum());
+		}
+
+		for ( int i = 0; i < result.size(); i++ ) {
+
+			GeneralDayDatum current = result.get(i);
+			JsonNode data = root.get("sunriseTimeUtc");
+			if ( data.isArray() ) {
+				current.setSunrise(parseTimeValue(data.get(i)));
+			}
+			data = root.get("sunsetTimeUtc");
+			if ( data.isArray() ) {
+				current.setSunset(parseTimeValue(data.get(i)));
+
+			}
+			data = root.get("moonriseTimeUtc");
+			if ( data.isArray() ) {
+				current.setMoonrise(parseTimeValue(data.get(i)));
+			}
+			data = root.get("moonsetTimeUtc");
+			if ( data.isArray() ) {
+				current.setMoonset(parseTimeValue(data.get(i)));
+			}
+			data = root.get("validTimeUtc");
+			if ( data.isArray() ) {
+				current.setCreated(new Date(data.get(i).asLong() * 1000));
+			}
+			data = root.get("temperatureMax");
+			if ( data.isArray() ) {
+				current.setTemperatureMaximum(parseBigDecimal(data.get(i)));
+			}
+			data = root.get("temperatureMin");
+			if ( data.isArray() ) {
+				current.setTemperatureMinimum(new BigDecimal(data.get(i).asLong()));
+			}
+
+		}
+
+		log.debug("Finished IBM Weather retrieval");
+		return result;
+	}
+
+	@Override
+	public Collection<WCHourlyDatum> readHourlyForecast(String locationIdentifier, String apiKey,
+			String datumPeriod) {
+		if ( locationIdentifier == null ) {
+			return null;
+		}
+		final List<WCHourlyDatum> result = new ArrayList<WCHourlyDatum>();
+		final String url = getURL(this.hourlyForecastUrl, locationIdentifier, apiKey, datumPeriod);
+		JsonNode root;
+		try {
+			log.debug("opening IBM Weather API connection");
+			log.debug(url);
```

Comment:
Rather than 2 debug logs, I'd rather see one, like
```java
log.debug("Connecting to IBM weather API {}", url);
```
Note the use of the argument placeholder.